### ISSUE TITLE
Error when running run_test.sh

### ISSUE THREAD

**Issue by wzmsltw** [facebookresearch/video-nonlocal-net#16]
Hi Xionglong

Since I failed to "git clone --recursive https://github.com/xiaolonw/caffe2".  I follow your instruction to replace ./caffe2/video files in official caffe2 to install.  When I running test_net_multicrop.sh, I met following error:

```
[INFO: checkpoints.py:  247]: res5_2_branch2c_bn_rm loaded from weights file into: gpu_0/res5_2_branch2c_bn_rm (2048,)
[INFO: checkpoints.py:  247]: res5_2_branch2c_bn_riv loaded from weights file into: gpu_0/res5_2_branch2c_bn_riv (2048,)
*** Aborted at 1525166815 (unix time) try "date -d @1525166815" if you are using GNU date ***
PC: @     0x7f7d8b793bd9 caffe2::CustomizedVideoInputOp<>::DecodeAndTransform()
*** SIGSEGV (@0x48) received by PID 25821 (TID 0x7f7d3eb9e700) from PID 72; stack trace: ***
    @     0x7f7d9bffa4b0 (unknown)
    @     0x7f7d8b793bd9 caffe2::CustomizedVideoInputOp<>::DecodeAndTransform()
    @     0x7f7d8b790c23 std::_Function_handler<>::_M_invoke()
    @     0x7f7d8b744d0b caffe2::TaskThreadPool::main_loop()
    @     0x7f7d9076cc5c execute_native_thread_routine_compat
    @     0x7f7d9caa66ba start_thread
    @     0x7f7d9c0cc41d clone
    @                0x0 (unknown)
Segmentation fault (core dumped)
```

It seems an error caused by customizedvideoinput, how can I correct it? Thanks so much!

**Comment by xuchao1688** [comment]
What does your test lmdb looks like? For train, each record in lmdb only has two items, the video path and label. However, in test, the record seems to have 3 or more items, guessing from `test_net_video.py`. However, your `create_video_lmdb_test_multicrop.py ` creates such test lmdb with some items lost and only two items left. The `vallist.txt` only has two items, the video path and label, too.

**Comment by xiaolonw** [comment]
So the testing lmdb actually specify the number of crops in temporal and number of crops in spatial. 

For example, 
https://github.com/facebookresearch/video-nonlocal-net/blob/master/process_data/kinetics/create_video_lmdb_test_multicrop.py#L34

here specifies we should crop the video for 10 times and we do 3 crops in spatial. The starting frame for testing for each clip is specified here:
https://github.com/facebookresearch/video-nonlocal-net/blob/master/caffe2_customized_ops/video/customized_video_input_op.h#L328

Note that in our implementation, the start_frm is ranging from 0 to 9 in the case of 10 temporal crops, and the actual starting frame is converted out in here: 
https://github.com/facebookresearch/video-nonlocal-net/blob/master/caffe2_customized_ops/video/customized_video_io.cc#L694

**Comment by wzmsltw** [comment]
Thanks for your quick reply!
So can I specify the start-frame of each clip in python script during  resting?

**Comment by xiaolonw** [comment]
No, I am afraid not ... you need to change the cpp files to do that ... the current testing scheme with cropping video 10 times will sample the video clip with 10 different starting frame. These 10 starting frames are sampled with equal time duration between every 2 starting frames. 

If you only crop the video 1 time, the starting frame will be always 0.
